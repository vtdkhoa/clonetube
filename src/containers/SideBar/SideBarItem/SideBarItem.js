import React from 'react'
import { Icon, Menu } from 'semantic-ui-react'
import { Link, withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import './SideBarItem.scss'

const SideBarItem = props => {
  const active = activeItem(props) ? 'active-item' : null

  return (
    <Link to={{ pathname: props.path }}>
      <Menu.Item className={['sidebar-item', active].join(' ')}>
        <div className="sidebar-item-alignment-container">
          <span>
            <Icon size="large" name={props.icon}/>
          </span>
          <span>{props.label}</span>
        </div>
      </Menu.Item>
    </Link>
  )
}

const activeItem = propsParam => {
  const { location, path } = propsParam
  const { pathname } = location

  if (path === '/') {
    return pathname === path
  }
  return pathname.includes(path)
}

SideBarItem.propTypes = {
  path: PropTypes.string,
  icon: PropTypes.string,
  label: PropTypes.string
}

export default withRouter(SideBarItem)