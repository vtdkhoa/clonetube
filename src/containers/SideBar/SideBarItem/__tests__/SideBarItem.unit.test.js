import React from 'react'
import SideBarItem from '../SideBarItem'
import { shallow } from 'enzyme'

describe('SideBarItem', () => {
  test('renders empty SideBarItem', () => {
    const wrapper = shallow(<SideBarItem/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders highlighted SideBarItem that navigates to /feed/trending', () => {
    const wrapper = shallow(
      <SideBarItem
        highlighted
        icon="fire"
        label="Trending"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('renders non-highlighted SideBarItem that navigated to /feed/trending', () => {
    const wrapper = shallow(
      <SideBarItem
        icon="fire"
        label="Trending"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('renders highlighted SideBarItem with no navigation', () => {
    const wrapper = shallow(
      <SideBarItem
        highlighted
        icon="fire"
        label="Trending"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})