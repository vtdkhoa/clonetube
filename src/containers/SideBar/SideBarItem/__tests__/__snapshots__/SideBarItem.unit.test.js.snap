// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem renders empty SideBarItem 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SideBarItem renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SideBarItem renders highlighted SideBarItem with no navigation 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SideBarItem renders non-highlighted SideBarItem that navigated to /feed/trending 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
